src/components/modals/ProfileManagementModal.vue

```diff
@@ -0,0 +1,77 @@
+<script lang="ts" setup>
+import ModalCard from '../ModalCard.vue';
+import { computed, getCurrentInstance, onMounted } from 'vue';
+import useStore from '../../store';
+import SettingsItem from '../settings-components/SettingsItem.vue';
+
+import VueRouter from 'vue-router';
+import R2Error from 'src/model/errors/R2Error';
+
+const store = useStore();
+let router!: VueRouter;
+
+onMounted(() => {
+    router = getCurrentInstance()!.proxy.$router;
+})
+
+const isOpen = computed(() => store.state.modals.isProfileManagementModalOpen);
+
+function closeModal() {
+    store.commit('closeProfileManagementModal');
+}
+
+async function changeProfile() {
+    store.commit("closeProfileManagementModal");
+    router.push({name: 'profiles'});
+}
+
+async function exportProfileAsFile() {
```

Comment:
Is it intentional that this leaves the modal open on error and success, whereas `exportProfileAsCode` closes the modal on error and switches to another modal on success?

Reply:
It is - The code export opens a new modal whereas this one doesn't and instead waits for the file selection dialog.

IMO it feels more natural to keep it like this - any thoughts?

Reply:
It's mostly just because Windows doesn't necessarily bring the file window to the foreground so the user doesn't really have any kind of confirmation that something has happened unless they happen to see File Explorer flashing.

We can probably circumvent this with a separate confirmation modal, although it feels redundant.

I'm happy to do a separate PR to auto-close it to keep it consistent though.